Kitodo/src/main/java/org/kitodo/production/forms/WorkflowForm.java

```diff
@@ -183,6 +189,26 @@ private void updateTemplateTasks() throws DAOException, IOException, WorkflowExc
         }
     }
 
+    /**
+     * Check if there are data editor setting for the tasks of the current workflow.
+     *
+     * @return true if one of the tasks has data editor settings defined
+     */
+    public boolean hasWorkflowDataEditorSettingsDefined() {
```

Comment:
Sorry for having another change request but I just noticed that you added this new function in the `WorkflowForm` class contains a whole bunch of loops and condition checks. Although we have many places where form classes/beans still contain a of such business logic, we are trying to avoid adding _new_ such cases to the beans. Could you instead just add the function to the `DataEditorSettingService` and save the return value to a flag in the form class? Than this function could simply return the determined value and it wouldn't have to be determined again each time the value is requested.

Reply:
I think it is not possible to store this value (session wide?). The value depends on the workflow which is currently opened in the Workflow editor (form). But I moved the logic to the service. In order to hopefully speed up the query i rewrote the check to HSQL.

Reply:
Thanks for the refactoring. 

Couldn't this new property (e.g. `boolean dataEditorSettingsDefined`) just be set once in `WorkflowForm.load` whenever the workflow editor is opened with a specific workflow? Thus the value would be evaluated _once_ every time a workflow is opened and the evaluated value would just be returned?

So something like this. First define new property of `WorkflowForm`: 
```
boolean dataEditorSettingsDefined;
```
then evaluate its value in the `load` method:
```
public void load(int id) {
...
dataEditorSettingsDefined = this.dataEditorSettingService.areDataEditorSettingsDefinedForWorkflow(this.workflow);
...
}
```
and then in the "getter" just return the evaluated value
```
public boolean hasWorkflowDataEditorSettingsDefined() {
   return dataEditorSettingsDefined;
}
```
Would that work?

Reply:
Thank you for the suggestion. I implemented that and it works.